### PR TITLE
Get rid of RDP false positives

### PR DESCRIPTION
**Pull request by 0xA50C1A1**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Describe changes:

Well, I ran into yet another problem: the RDP protocol dissector was giving false positives on [S7comm Plus](https://github.com/ntop/nDPI/files/13432228/s7comm-plus.zip) traffic. I've checked, false positives are caused by [this](https://github.com/ntop/nDPI/blob/21f25740337c6c0412181c135b8cee8881fc7f58/src/lib/protocols/rdp.c#L49C1-L54C128).

Only validation of TPKT and COTP headers is done here, and also checking if PDU type is a Connection Request. I refined this check to make it more stringent by simply adding a check to see if the current packet is an RDP Negotiation request or RDP Negotiation response.

### CODE REVIEW

File: src/lib/protocols/rdp.c [hunk 1]
```diff
@@ -46,17 +46,34 @@ static void ndpi_search_rdp(struct ndpi_detection_module_struct *ndpi_struct,
...
   if (packet->tcp != NULL) {
-    if (packet->payload_packet_len > 10
-	&& get_u_int8_t(packet->payload, 0) > 0
-	&& get_u_int8_t(packet->payload, 0) < 4 && get_u_int16_t(packet->payload, 2) == ntohs(packet->payload_packet_len)
-	&& get_u_int8_t(packet->payload, 4) == packet->payload_packet_len - 5
-	&& get_u_int8_t(packet->payload, 5) == 0xe0
-	&& get_u_int16_t(packet->payload, 6) == 0 && get_u_int16_t(packet->payload, 8) == 0 && get_u_int8_t(packet->payload, 10) == 0) {
-      ndpi_int_rdp_add_connection(ndpi_struct, flow);
+    /* Searching for a RDP Negotiation request from a client */
+    if(current_pkt_from_client_to_server(ndpi_struct, flow) &&
```
Comment (0xA50C1A1) [inline src/lib/protocols/rdp.c]: Well, perhaps `if(current_pkt_from_client_to_server(ndpi_struct, flow))` and `if(current_pkt_from_server_to_client(ndpi_struct, flow))` may be redundant.

**Comment by IvanNardi** [comment]
I am going to take a closer look, because I would like to avoid all these (performance) regressions, if possible

**Comment by 0xA50C1A1** [comment]
> I am going to take a closer look, because I would like to avoid all these (performance) regressions, if possible

Sure. Btw, false positives are now given by the H.323 dissector.  Detecting the protocol by TPKT and COTP headers only isn't a good idea, cuz there are a lot of protocols encapsulated in COTP. However, I guess I know how to fix it.

**Comment by IvanNardi** [comment]
@0xA50C1A1, what do you think of my change?

**Comment by 0xA50C1A1** [comment]
> @0xA50C1A1, what do you think of my change?

Yea, it got much better.

**Comment by IvanNardi** [comment]
> > @0xA50C1A1, what do you think of my change?
> 
> Yea, it got much better.

Do you think we can merge this PR?

**Comment by 0xA50C1A1** [comment]
> > > @0xA50C1A1, what do you think of my change?
> > 
> > 
> > Yea, it got much better.
> 
> Do you think we can merge this PR?

Yea. I tested this on Xrdp traffic and everything works.

**Review by IvanNardi — APPROVED** [review]

**Pull request merged**